shell/platform/darwin/ios/framework/Source/FlutterMetalLayerTest.mm

```diff
@@ -283,8 +283,11 @@ - (void)testDealloc {
     [compositor commitTransaction];
 
     [self removeMetalLayer:layer];
+  }
+  CFTimeInterval start = CACurrentMediaTime();
+  while (weakLayer != nil && CACurrentMediaTime() - start < 1) {
     // Deallocating the layer after removing is not synchronous.
-    CFRunLoopRunInMode(kCFRunLoopDefaultMode, 1, YES);
+    CFRunLoopRunInMode(kCFRunLoopDefaultMode, 0.01, YES);
```

Comment:
Would something like this work (I didn't try it):
```
@interface FlutterMetalLayerTest : XCTestCase
@property(weak) FlutterMetalLayer *weakLayer;
@end

- (void)testDealloc {
  @autoreleasepool {
...
    self.weakLayer = layer;
...
  }
  XCTestExpectation *weakNil = [self keyValueObservingExpectationForObject:self keyPath:@"weakLayer" expectedValue:nil];
  [self waitForExpectationsWithTimeout:1.0 handler:nil];
```

Reply:
(I'm not sure about the KVO part, but it would be nice to let `-waitForExpectations` handle the runloop spinning.

Reply:
@jmagman, I don't think KVO will work since zeroing weak reference doesn't trigger KVO notifications. I tried this
```objc
- (void)testDealloc {
   __weak FlutterMetalLayer* weakLayer;
  @autoreleasepool {
    FlutterMetalLayer* layer = [self addMetalLayer];
    weakLayer = layer;
    TestCompositor* compositor = [[TestCompositor alloc] initWithLayer:layer];

    id<CAMetalDrawable> drawable = [layer nextDrawable];
    BAIL_IF_NO_DRAWABLE(drawable);
    [drawable present];
    [compositor commitTransaction];

    [self removeMetalLayer:layer];
  }
  NSPredicate* predicate =
      [NSPredicate predicateWithBlock:^(id evaluatedObject, NSDictionary* bindings) {
        return weakLayer == nil;
      }];
  [self expectationForPredicate:predicate evaluatedWithObject:self handler:nil];
  [self waitForExpectationsWithTimeout:1 handler:nil];
}
```
Unfortunately it didn't work either. I can confirm that `weakLayer==nil` is `YES` in the predicate and yet the expection fails. Maybe I'm holding it wrong but I'm not sure how.

Reply:
> since zeroing weak reference doesn't trigger KVO notifications

That's what I suspected, but I couldn't think other another way.  Thanks for checking.